testing/soak/src/integTest/groovy/org/gradle/resolve/CachedMetadataRuleSoakTest.groovy

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.resolve
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+
+class CachedMetadataRuleSoakTest extends AbstractIntegrationSpec {
+    def "rule is cached across builds"() {
+        executer.requireOwnGradleUserHomeDir("cannot reuse cross-build caches")
+        buildFile << """
+        ${mavenCentralRepository()}
+
+        configurations {
+            conf
+        }
+
+        dependencies {
+            conf 'log4j:log4j:1.2.17'
+        }
+
+        task resolve(type: Sync) {
+            from configurations.conf
+            into 'libs'
+        }
+
+@CacheableRule
+class CachedRule implements ComponentMetadataRule {
+    public void execute(ComponentMetadataContext context) {
+            println "Rule executed on \${context.details.id}"
+            context.details.allVariants {
+                println("Variant \$it")
+                withDependencies { deps ->
+                    deps.each {
+                       println "See dependency: \$it"
```

Comment:
What actually gets printed here?

Seems like this dependency only has optional, provided, and test dependencies


https://mvnrepository.com/artifact/log4j/log4j/1.2.17

Reply:
Interesting

Reply:
Do you get the same behavior if you apply the java plugin?

Do we ever end up actually adding the JvmEcosystemVariantDerivationStrategy?

https://github.com/gradle/gradle/blob/020346f10f8d0b536a24e7736b20431e45812c0a/platforms/jvm/platform-jvm/src/main/java/org/gradle/api/plugins/JvmEcosystemPlugin.java#L65

Reply:
It looks the same